spec/models/post_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'Is valid with valid attributes' do
+    user = User.create(name: 'Aisha Mh.', posts_counter: 0)
+    post = user.posts.build(title: 'Post1', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
```

Comment:
- [ ] Please add a test for the custom method `increment_user_posts_counter` in the post model as requested in the project requirements to add unit specs for all of your models' methods.